### PR TITLE
Fixes from PVS Studio static analyzer

### PR DESCRIPTION
**Pull request by dimhotepus**
Some fixes from PVS Studio static analyzer.

Product page: https://www.viva64.com/en/pvs-studio/
How to use for free: https://www.viva64.com/en/b/0600/

### CODE REVIEW

**Comment by rmarquis** [comment]
Thanks!

Do you mind changing the categories of the commits, since `misc` is pretty much for the misc stuff only? See [wiki](https://github.com/etlegacy/etlegacy/wiki/How-to-commit-Your-Code#commit-category), f.e. use `cgame`, `ui`, `client`, ... and `general` if the changes affect multiple components.

**Comment by dimhotepus** [comment]
> Thanks!
> 
> Do you mind changing the categories of the commits, since `misc` is pretty much for the misc stuff only? See [wiki](https://github.com/etlegacy/etlegacy/wiki/How-to-commit-Your-Code#commit-category), f.e. use `cgame`, `ui`, `client`, ... and `general` if the changes affect multiple components.

I was thinking about misc as `code clean ups` from wiki. Should it be changed to exact touched subsystems?

**Comment by rmarquis** [comment]
Yes please.

**Comment by dimhotepus** [comment]
In progress, will publish soon.

File: src/renderer2/tr_shader.c [hunk 1]
```diff
@@ -1537,11 +1537,6 @@ qboolean LoadMap(shaderStage_t *stage, char *buffer)
 		stage->type = ST_LIGHTMAP;
 		return qtrue;
 	}
-	//skip normalmap on sky .... for now
-	else if (!Q_stricmp(token, "sky") )
-	{
-		imageBits |= !IF_NORMALMAP;
```
Comment (dimhotepus) [inline src/renderer2/tr_shader.c:1543]: `!IF_NORMALMAP` is `0`

**Comment by dimhotepus** [comment]
@rmarquis Could you, please, review :) ?

**Comment by rmarquis** [comment]
Sorry for the delay. I'm slowly getting through the review. It's mostly great, though there are a few commits I'm not too sure about (as I'm not familiar with some parts of the code).

In particular, the Memset changes look okay to me, but I'll do some more local testing to ensure nothing actually breaks. I'm also not familiar with SSE optimization, but that code isn't actually enabled (see #1442)

**Comment by dimhotepus** [comment]
Feel free to remove bad commits.

As for SSE support, it was a bug in `DotProduct`, no value was returned as a result of computation.

**Comment by rmarquis** [comment]
Rebased/merged, with very minor modification in some commit messages. Thanks a lot! And sorry it took so long.

**Comment by dimhotepus** [comment]
@rmarquis No problem, thank you for review and fixes!

**Pull request closed without merging**